recipes/stlab/all/conanfile.py

```diff
@@ -0,0 +1,65 @@
+from conans import ConanFile, tools
+import os
+
+class Stlab(ConanFile):
+    name = 'stlab'
+    description = 'The Software Technology Lab libraries.'
+    url = 'https://github.com/stlab/libraries'
+    homepage = 'https://github.com/stlab/libraries'
+    author = 'Manu Sanchez and Tim van Deurzen'
+    license = 'Boost Software License 1.0'
+    topics = 'c++', 'concurrency'
+
+    settings = 'compiler'
+
+    no_copy_source = True
+    _source_subfolder = 'source_subfolder'
+
+    requires = 'boost/1.69.0'
+    default_options = ('boost:shared=False',
+                       'boost:without_chrono=False',
+                       'boost:without_system=False',
+                       'boost:without_timer=False',
+                       'boost:without_test=False',
+                       'boost:without_iostreams=True',
+                       'boost:without_log=True',
+                       'boost:without_regex=True',
+                       'boost:without_locale=True',
+                       'boost:without_exception=True',
+                       'boost:without_filesystem=True',
+                       'boost:without_container=True',
+                       'boost:without_program_options=True',
+                       'boost:without_wave=True',
+                       'boost:without_thread=True',
+                       'boost:without_graph_parallel=True',
+                       'boost:without_context=True',
+                       'boost:without_random=True',
+                       'boost:without_graph=True',
+                       'boost:without_serialization=True',
+                       'boost:without_date_time=True',
+                       'boost:without_fiber=True',
+                       'boost:without_coroutine=True',
+                       'boost:without_mpi=True',
+                       'boost:without_type_erasure=True',
+                       'boost:without_math=True',
+                       'boost:without_container=True',
+                       'boost:without_log=True',
+                       'boost:without_exception=True',
+                       'boost:without_python=True',
+                       'boost:without_stacktrace=True',
+                       'boost:without_atomic=True')
```

Comment:
Is this really required?
It is up to the user to disable an option.

Reply:
I'm not 100% sure. It's the set of options listed in the original library `conanfile.txt`.

Reply:
You can remove these then.
Users can disable these options themselves.

Reply:
This is the struggle I always have when using monolithic frameworks with conan. Either explicitly set my exact dependencies, or leave all the work to the linker. The first option does not work very well with current conan CI workflows, the second one feels lazy and that can have problems on the user side.

Reply:
I don't understand what you mean.
By using the default options (and removing these), first-time users will be able to use the default boost package (built by CCI CI).
By overriding the defaults, a custom boost package is required.

Reply:
They will have a precompiled boost available, right, but they will pull and link all the boost libraries into their project, even those that are not needed.

Reply:
Oh yeah, now I understand your note about *monolithic*.
Users can *fix* this by using the `FindBoost` cmake module.
Soon, conan will support components.

See https://github.com/conan-io/conan/pull/6871 (and more)